tools/cli-application/src/main/kotlin/io/vrap/rmf/codegen/cli/CodegeneratorApp.kt

```diff
@@ -103,41 +108,66 @@ class GeneratorTask : Callable<Int> {
                 clientPackage = clientPackageName,
                 outputFolder = outputFolder
         )
-        val apiProvider = ApiProvider(ramlFileLocation)
 
-        val generatorComponent: GeneratorComponent = when (target) {
+        generate(ramlFileLocation, target, generatorConfig)
+        if (watch) {
+            val watchDir = ramlFileLocation.parent
+            val watcher = DirectoryWatcher.builder()
+                    .path(watchDir)
+                    .listener { event -> when(event.eventType()) {
+                        DirectoryChangeEvent.EventType.CREATE,
+                        DirectoryChangeEvent.EventType.MODIFY,
+                        DirectoryChangeEvent.EventType.DELETE -> {
+                            val json = event.path().toString().endsWith("json")
+                            val raml = event.path().toString().endsWith("raml")
+                            if (json || raml) {
+                                println("File ${event.eventType().name.toLowerCase()}: ${event.path()}")
+                                generate(ramlFileLocation, target, generatorConfig)
```

Comment:
If the spec is too large, this might cause too many jobs to stuck up in a queue, this can be with a framework like rxjava, by executing only the last job maybe.
WDYT?